File: meep_analyzer.py
"""
MEEP System Analyzer - Captura completa da estrutura do sistema MEEP
Usa Playwright para navegar e analisar o sistema real
"""

import asyncio
from playwright.async_api import async_playwright
import json
from datetime import datetime
import os

class MEEPAnalyzer:
    def __init__(self):
        self.base_url = "https://beta.portal.meep.com.br"
        self.email = "TORETOMAL@ICLOUD.COM"
        self.password = "10041210Cl@."
        self.modules_to_analyze = [
            # Módulos já confirmados ✅
            {"path": "/private/dashboard/general", "name": "Dashboard Geral"},
            {"path": "/private/cardapio/cardapios", "name": "Cardápios"},
            {"path": "/private/relatorios/venda", "name": "Relatórios de Venda"},
            {"path": "/private/cadastros/clientes", "name": "Cadastro de Clientes"},
            {"path": "/private/financeiro/conta-digital", "name": "Conta Digital"},
            {"path": "/private/pdv", "name": "PDV"},
            {"path": "/private/meeperp", "name": "MEEP ERP"},

            # Módulos a investigar 🔍
            {"path": "/private/gestao-venda", "name": "Gestão de Vendas"},
            {"path": "/private/solucoes-online", "name": "Soluções Online"},
            {"path": "/private/ingressos", "name": "Ingressos"},
            {"path": "/private/equipe", "name": "Equipe"},
            {"path": "/private/pedidos", "name": "Pedidos"},
            {"path": "/private/mapa-operacao", "name": "Mapa de Operação"},
            {"path": "/private/marketing", "name": "Marketing"},
            {"path": "/private/bi", "name": "Business Intelligence"},
            {"path": "/private/automacao", "name": "Automação"},
            {"path": "/private/integracao", "name": "Integração"},
        ]
        self.captured_data = {
            "timestamp": datetime.now().isoformat(),
            "modules": [],
            "api_endpoints": [],
            "forms": [],
            "tables": [],
            "navigation": []
        }

    async def analyze(self):
        async with async_playwright() as p:
            browser = await p.chromium.launch(headless=False)
            context = await browser.new_context()
            page = await context.new_page()

            # Interceptar requisições de API
            page.on("request", self.capture_api_request)
            page.on("response", self.capture_api_response)

            try:
                # 1. Login
                print("🔐 Fazendo login no MEEP...")
                await self.login(page)

                # 2. Analisar cada módulo
                for module in self.modules_to_analyze:
                    print(f"📊 Analisando módulo: {module['name']}...")
                    await self.analyze_module(page, module)

                # 3. Capturar menu lateral completo
                print("🗂️ Capturando estrutura de navegação...")
                await self.capture_navigation(page)

                # 4. Salvar resultados
                await self.save_results()

            except Exception as e:
                print(f"❌ Erro: {e}")
            finally:
                await browser.close()

    async def login(self, page):
        """Realiza login no sistema MEEP"""
        await page.goto(self.base_url + "/login")
        await page.wait_for_load_state("networkidle")

        # Preencher formulário
        await page.fill('input[type="email"]', self.email)
        await page.fill('input[type="password"]', self.password)

        # Clicar no botão de login
        await page.click('button[type="submit"]')

        # Aguardar redirecionamento
        await page.wait_for_url("**/private/**", timeout=10000)
        print("✅ Login realizado com sucesso!")

    async def analyze_module(self, page, module):
        """Analisa um módulo específico"""
        try:
            await page.goto(self.base_url + module["path"])
            await page.wait_for_load_state("networkidle")
            await asyncio.sleep(2)  # Aguardar carregamento completo

            module_data = {
                "name": module["name"],
                "path": module["path"],
                "elements": {
                    "forms": [],
                    "tables": [],
                    "buttons": [],
                    "filters": [],
                    "charts": []
                }
            }

            # Capturar formulários
            forms = await page.query_selector_all('form')
            for form in forms:
                form_data = await self.extract_form_data(form)
                if form_data:
                    module_data["elements"]["forms"].append(form_data)

            # Capturar tabelas
            tables = await page.query_selector_all('table')
            for table in tables:
                table_data = await self.extract_table_structure(table)
                if table_data:
                    module_data["elements"]["tables"].append(table_data)

            # Capturar botões de ação
            buttons = await page.query_selector_all('button')
            for button in buttons:
                button_text = await button.text_content()
                if button_text:
                    module_data["elements"]["buttons"].append(button_text.strip())

            # Capturar filtros
            filters = await page.query_selector_all('[class*="filter"], [class*="search"]')
            for filter_el in filters:
                filter_type = await filter_el.get_attribute("type")
                module_data["elements"]["filters"].append(filter_type)

            # Capturar gráficos (canvas, svg com charts)
            charts = await page.query_selector_all('canvas, svg[class*="chart"]')
            module_data["elements"]["charts"] = len(charts)

            self.captured_data["modules"].append(module_data)
            print(f"  ✅ Módulo {module['name']} analisado")

        except Exception as e:
            print(f"  ⚠️ Erro ao analisar {module['name']}: {e}")

    async def extract_form_data(self, form):
        """Extrai estrutura de um formulário"""
        try:
            inputs = await form.query_selector_all('input, select, textarea')
            fields = []
            for input_el in inputs:
                field = {
                    "name": await input_el.get_attribute("name"),
                    "type": await input_el.get_attribute("type"),
                    "required": await input_el.get_attribute("required"),
                }
                fields.append(field)
            return {"fields": fields, "count": len(fields)}
        except:
            return None

    async def extract_table_structure(self, table):
        """Extrai estrutura de uma tabela"""
        try:
            headers = await table.query_selector_all('th')
            columns = []
            for header in headers:
                text = await header.text_content()
                if text:
                    columns.append(text.strip())

            rows = await table.query_selector_all('tbody tr')
            return {
                "columns": columns,
                "row_count": len(rows)
            }
        except:
            return None

    async def capture_navigation(self, page):
        """Captura estrutura completa do menu de navegação"""
        try:
            # Procurar menu lateral
            menu_items = await page.query_selector_all('[class*="menu-item"], [class*="nav-item"], aside a')

            for item in menu_items:
                text = await item.text_content()
                href = await item.get_attribute("href")
                if text and href:
                    self.captured_data["navigation"].append({
                        "text": text.strip(),
                        "href": href
                    })
        except Exception as e:
            print(f"  ⚠️ Erro ao capturar navegação: {e}")

    def capture_api_request(self, request):
        """Captura requisições de API"""
        if "/api/" in request.url or "/private/api/" in request.url:
            self.captured_data["api_endpoints"].append({
                "url": request.url,
                "method": request.method,
                "timestamp": datetime.now().isoformat()
            })

    def capture_api_response(self, response):
        """Captura respostas de API"""
        # Podemos adicionar análise de respostas se necessário
        pass

    async def save_results(self):
        """Salva resultados da análise"""
        filename = f"meep_analysis_{datetime.now().strftime('%Y%m%d_%H%M%S')}.json"
        with open(filename, 'w', encoding='utf-8') as f:
            json.dump(self.captured_data, f, indent=2, ensure_ascii=False)

        print(f"\n📁 Análise salva em: {filename}")
        print(f"📊 Módulos analisados: {len(self.captured_data['modules'])}")
        print(f"🔗 Endpoints capturados: {len(self.captured_data['api_endpoints'])}")
        print(f"🧭 Items de navegação: {len(self.captured_data['navigation'])}")

async def main():
    print("🚀 Iniciando análise do sistema MEEP...")
    analyzer = MEEPAnalyzer()
    await analyzer.analyze()
    print("✅ Análise completa!")

if __name__ == "__main__":
    # Instalar playwright se necessário
    os.system("pip install playwright")
    os.system("playwright install chromium")

    # Executar análise
    asyncio.run(main())

File: paineluniversal/paineluniversal/backend/app/copiloto sp trab/meep_analyzer.js
const puppeteer = require('puppeteer');
const fs = require('fs').promises;
const path = require('path');

class MEEPAnalyzer {
    constructor() {
        this.baseUrl = 'https://beta.portal.meep.com.br';
        this.credentials = {
            email: 'toretomal@icloud.com',
            password: '10041210Cl@'
        };
        this.analysisData = {
            timestamp: new Date().toISOString(),
            modules: {},
            dataStructures: {},
            workflows: {},
            apis: [],
            integrations: [],
            permissions: {}
        };
    }

    async initialize() {
        console.log('🚀 Iniciando análise do sistema MEEP...');
        this.browser = await puppeteer.launch({
            headless: false,
            defaultViewport: null,
            args: [
                '--no-sandbox',
                '--disable-setuid-sandbox',
                '--disable-dev-shm-usage',
                '--disable-accelerated-2d-canvas',
                '--disable-gpu'
            ]
        });
        this.page = await this.browser.newPage();
        
        // Interceptar requisições de rede
        await this.page.setRequestInterception(true);
        
        this.page.on('request', request => {
            const url = request.url();
            if (url.includes('/api/') || url.includes('/v1/') || url.includes('/v2/')) {
                this.captureAPIEndpoint(request);
            }
            request.continue();
        });

        this.page.on('response', response => {
            const url = response.url();
            if (url.includes('/api/') || url.includes('/v1/') || url.includes('/v2/')) {
                this.captureAPIResponse(response);
            }
        });

        await this.page.goto(this.baseUrl, { waitUntil: 'networkidle2' });
    }

    captureAPIEndpoint(request) {
        const endpoint = {
            url: request.url(),
            method: request.method(),
            headers: request.headers(),
            postData: request.postData(),
            timestamp: new Date().toISOString()
        };
        
        if (!this.analysisData.apis.find(api => api.url === endpoint.url && api.method === endpoint.method)) {
            this.analysisData.apis.push(endpoint);
            console.log(`📡 API capturada: ${endpoint.method} ${endpoint.url}`);
        }
    }

    async captureAPIResponse(response) {
        try {
            const url = response.url();
            const status = response.status();
            
            if (status === 200) {
                const contentType = response.headers()['content-type'];
                if (contentType && contentType.includes('application/json')) {
                    const data = await response.json().catch(() => null);
                    if (data) {
                        this.analyzeDataStructure(url, data);
                    }
                }
            }
        } catch (error) {
            // Silently handle errors
        }
    }

    analyzeDataStructure(url, data) {
        const endpoint = url.replace(this.baseUrl, '');
        const structure = this.extractStructure(data);
        
        if (!this.analysisData.dataStructures[endpoint]) {
            this.analysisData.dataStructures[endpoint] = structure;
            console.log(`📊 Estrutura de dados capturada: ${endpoint}`);
        }
    }

    extractStructure(obj, depth = 0) {
        if (depth > 5) return 'object';
        
        if (Array.isArray(obj)) {
            if (obj.length > 0) {
                return [this.extractStructure(obj[0], depth + 1)];
            }
            return [];
        }
        
        if (obj && typeof obj === 'object') {
            const structure = {};
            for (const key in obj) {
                const value = obj[key];
                structure[key] = typeof value === 'object' ? 
                    this.extractStructure(value, depth + 1) : 
                    typeof value;
            }
            return structure;
        }
        
        return typeof obj;
    }

    async login() {
        console.log('🔐 Realizando login...');
        
        try {
            // Aguardar o formulário de login carregar
            await this.page.waitForSelector('input[type="email"], input[type="text"], input[name="email"], input[name="username"]', { timeout: 10000 });
            
            // Identificar campos de login
            const emailField = await this.page.$('input[type="email"], input[type="text"], input[name="email"], input[name="username"], input[placeholder*="mail"], input[placeholder*="usuário"]');
            const passwordField = await this.page.$('input[type="password"], input[name="password"], input[name="senha"], input[placeholder*="senha"], input[placeholder*="password"]');
            
            if (emailField && passwordField) {
                await emailField.type(this.credentials.email, { delay: 50 });
                await passwordField.type(this.credentials.password, { delay: 50 });
                
                // Procurar botão de submit
                const submitButton = await this.page.$('button[type="submit"], input[type="submit"], button:has-text("Entrar"), button:has-text("Login"), button:has-text("Acessar")');
                
                if (submitButton) {
                    await Promise.all([
                        this.page.waitForNavigation({ waitUntil: 'networkidle2' }),
                        submitButton.click()
                    ]);
                    
                    console.log('✅ Login realizado com sucesso!');
                    await this.page.waitForTimeout(3000);
                    return true;
                }
            }
            
            console.log('❌ Não foi possível identificar os campos de login');
            return false;
        } catch (error) {
            console.error('❌ Erro no login:', error.message);
            return false;
        }
    }

    async analyzeModule(moduleName, selector) {
        console.log(`\n📦 Analisando módulo: ${moduleName}`);
        
        try {
            const moduleData = {
                name: moduleName,
                timestamp: new Date().toISOString(),
                elements: [],
                forms: [],
                tables: [],
                buttons: [],
                links: []
            };

            // Navegar para o módulo se houver selector
            if (selector) {
                const element = await this.page.$(selector);
                if (element) {
                    await element.click();
                    await this.page.waitForTimeout(2000);
                }
            }

            // Capturar elementos da página
            moduleData.elements = await this.page.evaluate(() => {
                const elements = [];
                
                // Inputs
                document.querySelectorAll('input, textarea, select').forEach(el => {
                    elements.push({
                        type: 'input',
                        tag: el.tagName.toLowerCase(),
                        name: el.name || el.id,
                        inputType: el.type,
                        placeholder: el.placeholder,
                        required: el.required,
                        pattern: el.pattern,
                        maxLength: el.maxLength
                    });
                });
                
                // Buttons
                document.querySelectorAll('button, a.btn, input[type="submit"]').forEach(el => {
                    elements.push({
                        type: 'button',
                        text: el.textContent.trim(),
                        href: el.href,
                        onclick: el.onclick ? 'has-handler' : null
                    });
                });
                
                // Tables
                document.querySelectorAll('table').forEach(table => {
                    const headers = Array.from(table.querySelectorAll('th')).map(th => th.textContent.trim());
                    elements.push({
                        type: 'table',
                        headers: headers,
                        rowCount: table.querySelectorAll('tbody tr').length
                    });
                });
                
                return elements;
            });

            this.analysisData.modules[moduleName] = moduleData;
            console.log(`✅ Módulo ${moduleName} analisado: ${moduleData.elements.length} elementos encontrados`);
            
        } catch (error) {
            console.error(`❌ Erro ao analisar módulo ${moduleName}:`, error.message);
        }
    }

    async analyzeAllModules() {
        const modules = [
            { name: 'Dashboard', selector: 'a[href*="dashboard"], nav a:has-text("Dashboard"), button:has-text("Dashboard")' },
            { name: 'Eventos/Caixa', selector: 'a[href*="evento"], a[href*="caixa"], nav a:has-text("Evento"), nav a:has-text("Caixa")' },
            { name: 'Clientes', selector: 'a[href*="cliente"], nav a:has-text("Cliente"), nav a:has-text("Informações")' },
            { name: 'Equipe', selector: 'a[href*="equipe"], a[href*="colaborador"], nav a:has-text("Equipe")' },
            { name: 'Cardápio', selector: 'a[href*="cardapio"], a[href*="produto"], nav a:has-text("Cardápio")' },
            { name: 'Vendas', selector: 'a[href*="venda"], nav a:has-text("Venda"), nav a:has-text("Gestão de venda")' },
            { name: 'Soluções Online', selector: 'a[href*="online"], a[href*="ingresso"], nav a:has-text("Soluções Online")' },
            { name: 'Relatórios', selector: 'a[href*="relatorio"], nav a:has-text("Relatório")' },
            { name: 'Estoque', selector: 'a[href*="estoque"], nav a:has-text("Estoque"), nav a:has-text("Gestão de estoque")' },
            { name: 'PDV', selector: 'a[href*="pdv"], nav a:has-text("PDV"), nav a:has-text("Ponto de Venda")' },
            { name: 'Equipamentos', selector: 'a[href*="equipamento"], a[href*="impressora"], nav a:has-text("Equipamento")' },
            { name: 'Pedidos', selector: 'a[href*="pedido"], nav a:has-text("Pedido"), nav a:has-text("Gestor de pedidos")' },
            { name: 'Financeiro', selector: 'a[href*="financeiro"], nav a:has-text("Financeiro")' },
            { name: 'Pagamentos', selector: 'a[href*="pagamento"], nav a:has-text("Pagamento")' },
            { name: 'Split', selector: 'a[href*="split"], nav a:has-text("Split")' },
            { name: 'Mapa Operação', selector: 'a[href*="mapa"], nav a:has-text("Mapa")' },
            { name: 'Mesas/Comandas', selector: 'a[href*="mesa"], a[href*="comanda"], nav a:has-text("Mesa")' },
            { name: 'Cartões', selector: 'a[href*="cartao"], nav a:has-text("Cartão"), nav a:has-text("Pré-ativação")' },
            { name: 'Marketing', selector: 'a[href*="marketing"], a[href*="fidelidade"], nav a:has-text("Marketing")' },
            { name: 'Cupons', selector: 'a[href*="cupom"], a[href*="desconto"], nav a:has-text("Cupom")' },
            { name: 'BI', selector: 'a[href*="bi"], a[href*="analytics"], nav a:has-text("BI"), nav a:has-text("Business Intelligence")' },
            { name: 'ERP', selector: 'a[href*="erp"], nav a:has-text("ERP")' },
            { name: 'Automações', selector: 'a[href*="automacao"], a[href*="integracao"], nav a:has-text("Automação")' }
        ];

        for (const module of modules) {
            await this.analyzeModule(module.name, module.selector);
            await this.page.waitForTimeout(1000);
        }
    }

    async captureMenuStructure() {
        console.log('\n🗂️ Capturando estrutura do menu...');
        
        const menuStructure = await this.page.evaluate(() => {
            const menus = [];
            
            // Procurar por diferentes tipos de menu
            const menuSelectors = [
                'nav ul li',
                '.sidebar ul li',
                '.menu ul li',
                '[role="navigation"] a',
                '.nav-menu a'
            ];
            
            menuSelectors.forEach(selector => {
                document.querySelectorAll(selector).forEach(item => {
                    const text = item.textContent.trim();
                    const href = item.querySelector('a')?.href || item.href;
                    if (text && !menus.find(m => m.text === text)) {
                        menus.push({ text, href });
                    }
                });
            });
            
            return menus;
        });
        
        this.analysisData.menuStructure = menuStructure;
        console.log(`✅ Estrutura do menu capturada: ${menuStructure.length} itens`);
    }

    async captureScreenshot(name) {
        const screenshotPath = path.join(__dirname, 'screenshots', `${name}_${Date.now()}.png`);
        await this.page.screenshot({ path: screenshotPath, fullPage: true });
        console.log(`📸 Screenshot salva: ${screenshotPath}`);
    }

    async saveAnalysis() {
        const outputPath = path.join(__dirname, `meep_analysis_${Date.now()}.json`);
        await fs.writeFile(outputPath, JSON.stringify(this.analysisData, null, 2));
        console.log(`\n💾 Análise salva em: ${outputPath}`);
        
        // Gerar relatório markdown
        const report = this.generateMarkdownReport();
        const reportPath = path.join(__dirname, `MEEP_REVERSE_ENGINEERING.md`);
        await fs.writeFile(reportPath, report);
        console.log(`📄 Relatório gerado: ${reportPath}`);
    }

    generateMarkdownReport() {
        let report = `# MEEP - Engenharia Reversa Completa
        
## 📅 Data da Análise: ${this.analysisData.timestamp}

## 🌐 URL Base: ${this.baseUrl}

## 📊 Resumo da Análise

- **Módulos Analisados**: ${Object.keys(this.analysisData.modules).length}
- **APIs Capturadas**: ${this.analysisData.apis.length}
- **Estruturas de Dados**: ${Object.keys(this.analysisData.dataStructures).length}

---

## 🗂️ Estrutura de Módulos

`;

        // Módulos
        Object.entries(this.analysisData.modules).forEach(([name, data]) => {
            report += `\n### 📦 ${name}\n\n`;
            report += `**Elementos Capturados**: ${data.elements.length}\n\n`;
            
            // Inputs
            const inputs = data.elements.filter(e => e.type === 'input');
            if (inputs.length > 0) {
                report += `#### Campos de Entrada\n\n`;
                inputs.forEach(input => {
                    report += `- **${input.name || 'unnamed'}**: ${input.inputType} ${input.required ? '(obrigatório)' : ''}\n`;
                    if (input.placeholder) report += `  - Placeholder: "${input.placeholder}"\n`;
                    if (input.pattern) report += `  - Pattern: \`${input.pattern}\`\n`;
                });
                report += '\n';
            }
            
            // Tables
            const tables = data.elements.filter(e => e.type === 'table');
            if (tables.length > 0) {
                report += `#### Tabelas\n\n`;
                tables.forEach(table => {
                    report += `- **Colunas**: ${table.headers.join(', ')}\n`;
                    report += `  - Linhas: ${table.rowCount}\n`;
                });
                report += '\n';
            }
        });

        // APIs
        if (this.analysisData.apis.length > 0) {
            report += `\n## 🔌 APIs e Endpoints\n\n`;
            
            const groupedApis = {};
            this.analysisData.apis.forEach(api => {
                const path = new URL(api.url).pathname;
                const base = path.split('/').slice(0, 3).join('/');
                if (!groupedApis[base]) groupedApis[base] = [];
                groupedApis[base].push(api);
            });
            
            Object.entries(groupedApis).forEach(([base, apis]) => {
                report += `### ${base}\n\n`;
                apis.forEach(api => {
                    const path = new URL(api.url).pathname;
                    report += `- **${api.method}** \`${path}\`\n`;
                });
                report += '\n';
            });
        }

        // Estruturas de Dados
        if (Object.keys(this.analysisData.dataStructures).length > 0) {
            report += `\n## 📋 Estruturas de Dados\n\n`;
            
            Object.entries(this.analysisData.dataStructures).forEach(([endpoint, structure]) => {
                report += `### ${endpoint}\n\n\`\`\`json\n${JSON.stringify(structure, null, 2)}\n\`\`\`\n\n`;
            });
        }

        return report;
    }

    async close() {
        if (this.browser) {
            await this.browser.close();
        }
    }

    async run() {
        try {
            await this.initialize();
            
            const loggedIn = await this.login();
            
            if (loggedIn) {
                await this.captureScreenshot('dashboard');
                await this.captureMenuStructure();
                await this.analyzeAllModules();
            }
            
            await this.saveAnalysis();
            
        } catch (error) {
            console.error('❌ Erro durante análise:', error);
        } finally {
            await this.close();
        }
    }
}

// Executar análise
(async () => {
    const analyzer = new MEEPAnalyzer();
    await analyzer.run();
})();